plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/lightgrid/LightGrid.java

```diff
@@ -671,6 +671,9 @@ public GridCell posToCell(GridPos pos)
         if (pos.col < 0 || pos.row < 0) {
             return null;
         }
+        if (pos.col >= columnElements.length || pos.row >= rowElements.length) {
+            return null;
+        }
```

Comment:
Please add sanity check that returned _cell_ is not null at:
https://github.com/dbeaver/dbeaver/blob/b4e57852deeeb175ca6ffb43357d23fa13c2ec1e/plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/lightgrid/LightGrid.java#L4256-L4258
```

Reply:
Thnaks for the insight @ShadelessFox. Do you know if there are other sanity checks to do? The method seams to be used at many places... Maybe I should actually revert my changes and put the out of bounds check at a higher level in the code to avoid breaking something else?

Reply:
If you're using Intellij IDEA, you can highlight method and press `Alt+F7` to find all the usages of it. I've done it already, so this is the only place where null check is missing.
This method is marked as _nullable_ and there's a similar check above, so everything is OK. Actually, your PR is self-contained, you didn't break anything, but adding this little sanity check would be nice.

Reply:
Ok thanks :) then if you're confident I'll just add the extra sanity check you're requesting. I double checked the method usages and you seam to be right, there is already a null check everywhere else.

Reply:
Done